AwsCryptographicMaterialProviders/runtimes/python/src/aws_cryptographic_materialproviders/smithygenerated/aws_cryptography_keystore/client.py

```diff
@@ -0,0 +1,462 @@
+# Code generated by smithy-python-codegen DO NOT EDIT.
+
+import asyncio
+from asyncio import sleep
+from copy import deepcopy
+from software_amazon_cryptography_keystore_internaldafny_types import IKeyStoreClient
+from typing import Awaitable, Callable, TypeVar, cast
+
+from .config import Config, KeyStoreConfig
+from .dafny_protocol import DafnyRequest, DafnyResponse
+from .plugin import set_config_impl
+from smithy_python.exceptions import SmithyRetryException
+from smithy_python.interfaces.interceptor import Interceptor, InterceptorContext
+from smithy_python.interfaces.retries import RetryErrorInfo, RetryErrorType
+
+from .config import Plugin
+from .deserialize import (
+    _deserialize_create_key,
+    _deserialize_create_key_store,
+    _deserialize_get_active_branch_key,
+    _deserialize_get_beacon_key,
+    _deserialize_get_branch_key_version,
+    _deserialize_get_key_store_info,
+    _deserialize_version_key,
+)
+from .errors import ServiceError
+from .models import (
+    CreateKeyInput,
+    CreateKeyOutput,
+    CreateKeyStoreInput,
+    CreateKeyStoreOutput,
+    GetActiveBranchKeyInput,
+    GetActiveBranchKeyOutput,
+    GetBeaconKeyInput,
+    GetBeaconKeyOutput,
+    GetBranchKeyVersionInput,
+    GetBranchKeyVersionOutput,
+    GetKeyStoreInfoOutput,
+    Unit,
+    VersionKeyInput,
+    VersionKeyOutput,
+)
+from .serialize import (
+    _serialize_create_key,
+    _serialize_create_key_store,
+    _serialize_get_active_branch_key,
+    _serialize_get_beacon_key,
+    _serialize_get_branch_key_version,
+    _serialize_get_key_store_info,
+    _serialize_version_key,
+)
+
+
+Input = TypeVar("Input")
+Output = TypeVar("Output")
+
+
+class KeyStore:
+    """Client for KeyStore
+
+    :param config: Configuration for the client.
+    """
+
+    def __init__(
+        self,
+        config: KeyStoreConfig | None = None,
+        dafny_client: IKeyStoreClient | None = None,
```

Comment:
Do we need to expose this internal dafny class to the customers?

Reply:
No, but Python classes are only allowed one `__init__` method. There may be some way to refactor this to create a static method on generated clients like `from_dafny_client".

Reply:
I think with the whole outer-module stuff, customers might not have access (shouldn't access) to the internal dafny stuff.

Reply:
Correct, they shouldn't do this, even regardless of outer-module changes.

Reply:
Ahh that's where `None` type comes in play. Got it.